httpbakery/client_test.go

```diff
@@ -156,6 +156,45 @@ func (s *ClientSuite) TestMacaroonCookiePath(c *gc.C) {
 	assertCookieCount("/foo/bar/baz", 1)
 }
 
+func (s *ClientSuite) TestPublicKey(c *gc.C) {
+	d := bakerytest.NewDischarger(nil, noCaveatChecker)
+	defer d.Close()
+	client := httpbakery.NewHTTPClient()
+	publicKey, err := httpbakery.PublicKeyForLocation(client, d.Location())
+	c.Assert(err, gc.IsNil)
+	expectedKey := d.Service.PublicKey()
+	c.Assert(publicKey, gc.DeepEquals, expectedKey)
+}
+
+func (s *ClientSuite) TestPublicKeyWrongURL(c *gc.C) {
+	client := httpbakery.NewHTTPClient()
+	_, err := httpbakery.PublicKeyForLocation(client, "http://localhost:0")
+	c.Assert(err, gc.ErrorMatches,
+		`cannot get public key from "http://localhost:0/publickey": Get http://localhost:0/publickey: dial tcp 127.0.0.1:0: connection refused`)
```

Comment:
OK, so it seems that the url in our annotation is redundant. Let's leave it out.